beacon-chain/core/helpers/randao.go

```diff
@@ -80,5 +80,6 @@ func RandaoMix(state *pb.BeaconState, wantedEpoch uint64) ([]byte, error) {
 		return nil, fmt.Errorf("input randaoMix epoch %d out of bounds: %d <= epoch < %d",
 			wantedEpoch, earliestEpoch, currentEpoch)
 	}
-	return state.LatestRandaoMixes[wantedEpoch%params.BeaconConfig().LatestRandaoMixesLength], nil
+	idx := wantedEpoch % params.BeaconConfig().LatestRandaoMixesLength
```

Comment:
check if the index is valid here, basically `if idx >- len(state.LatestRandaoMixes) `

Reply:
added test and if

Reply:
i dont think you pushed your latest changes up